lib/segment.ex

```diff
@@ -1,14 +1,12 @@
 defmodule Segment do
+  require Logger
+  @service Application.get_env(:segment, :sender_impl, Segment.Analytics.Batcher)
 
-  @type status :: :ok | :error
-
-  @spec start_link(binary) :: { Segment.status, pid }
-  def start_link(write_key) do
-    Agent.start_link(fn -> write_key end, name: __MODULE__)
+  def start_link(api_key) do
```

Comment:
We should add docs to these as well as some documentation at the module level here for how to configure

Reply:
yeah docs and moduledocs to come